### ISSUE TITLE
[New rule] `ban-name` Ban some user-specified functions (Willing to PR)

### ISSUE THREAD

**Issue by fzyzcjy** [dart-code-checker/dart-code-metrics#702]
**Please describe what the rule should do:**
The user can configure some functions that he wants to ban. For example, in my own app, I want to ban `showDialog` and force to use `myShowDialog` instead. This is because (let me explain the simplified case) I added logging in `myShowDialog`, and I do want every dialog to be logged.

**If your rule is inspired by other please provide link to it:**
N/A

**What category of rule is this? (place an "X" next to just one item)**

[x] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about (it will be better if you can provide both good and bad examples):**

configuration:

```
ban_function:
  showDialog: 'Please use myShowDialog instead'
  ... # other functions to ban
```

```dart
showDialog(...); // LINT WARN!

myShowDialog(...); // good
```

**Are you willing to submit a pull request to implement this rule?**
yes

**Comment by fzyzcjy** [comment]
#708 is ready for review now

**Comment by dkrutskikh** [comment]
Available in 4.12.0 release 🚀
